system/mount.py

```diff
@@ -250,87 +344,138 @@ def unset_mount(module, **kwargs):
+    # solaris kwargs:
+    #   name, src, fstype, opts, boot, passno, state, fstab=/etc/vfstab
+    # linux kwargs:
+    #   name, src, fstype, opts, dump, passno, state, fstab=/etc/fstab
     if get_platform() == 'SunOS':
         args = dict(
-            opts   = '-',
-            passno = '-',
-            fstab  = '/etc/vfstab',
-            boot   = 'yes'
+            opts='-',
+            passno='-',
+            fstab='/etc/vfstab',
+            boot='yes'
         )
     else:
         args = dict(
-            opts   = 'default',
-            dump   = '0',
-            passno = '0',
-            fstab  = '/etc/fstab'
+            opts='default',
+            dump='0',
+            passno='0',
+            fstab='/etc/fstab'
         )
     args.update(kwargs)
 
-    mount_bin = module.get_bin_path('mount')
-
+    mount_bin = module.get_bin_path('mount', required=True)
     name = kwargs['name']
-
-    cmd = [ mount_bin, ]
+    cmd = [mount_bin]
 
     if ismount(name):
-        cmd += [ '-o', 'remount', ]
+        cmd += ['-o', 'remount']
 
     if get_platform().lower() == 'freebsd':
-        cmd += [ '-F', args['fstab'], ]
-
-    if get_platform().lower() == 'linux':
-        cmd += [ '-T', args['fstab'], ]
+        cmd += ['-F', args['fstab']]
+    elif get_platform().lower() == 'linux':
+        cmd += ['-T', args['fstab']]
 
-    cmd += [ name, ]
+    cmd += [name]
 
     rc, out, err = module.run_command(cmd)
+
     if rc == 0:
         return 0, ''
     else:
         return rc, out+err
 
+
 def umount(module, **kwargs):
-    """ unmount a path """
+    """Unmount a path."""
 
-    umount_bin = module.get_bin_path('umount')
+    umount_bin = module.get_bin_path('umount', required=True)
     name = kwargs['name']
     cmd = [umount_bin, name]
 
     rc, out, err = module.run_command(cmd)
+
     if rc == 0:
         return 0, ''
     else:
         return rc, out+err
 
-def main():
 
+def bind_ismount(module, **kwargs):
+    """Check if bind mount point is mounted."""
+
+    if 'bind' not in kwargs.get('opts', []):
+        return False
+
+    changed = True
+    bin_path = module.get_bin_path('mount', required=True)
+    cmd = '%s -l' % bin_path
+
+    if get_platform().lower() == 'linux':
+        bin_path = module.get_bin_path('findmnt', required=True)
+        cmd = '%s -nr %s' % (bin_path, kwargs['name'])
+
+    rc, out, err = module.run_command(cmd)
+    mounts = []
+
+    if len(out):
+        mounts = out.strip().split('\n')
+
+    pattern = re.compile('\[(.*)\]')
+
+    for mnt in mounts:
+        arguments = mnt.split()
+
+        if get_platform().lower() == 'linux':
+            source = arguments[1]
+            result = pattern.search(arguments[1])
+
+            if len(result.groups()) > 0:
+                source = result.group(1)
+
+            if arguments[0] == kwargs['name'] and source == kwargs['src']:
+                changed = False
+        elif (
```

Comment:
Personally not a fan of the open paren like this, but I'll leave that up to the module maintainers ;->